src/bootstrap/check.rs

```diff
@@ -510,7 +510,9 @@ tool_check_step!(Rls, "src/tools/rls", SourceType::InTree);
 tool_check_step!(Rustfmt, "src/tools/rustfmt", SourceType::InTree);
 tool_check_step!(MiroptTestTools, "src/tools/miropt-test-tools", SourceType::InTree);
 
+// FIXME: currently  these are marked as ToolRustc, but they should be ToolBootstrap instead to avoid having to build the compiler first
 tool_check_step!(Bootstrap, "src/bootstrap", SourceType::InTree, false);
+tool_check_step!(BootstrapShim, "src/bootstrap/bin/bootstrap-shim", SourceType::InTree, false);
```

Comment:
This feels off to me - shouldn't checking bootstrap be enough? I would expect that we only need a single check target for each Cargo package, not for each binary target...

Reply:
> This feels off to me - shouldn't checking bootstrap be enough?

No, sadly not enough. The same approach is applied for miri as well:
https://github.com/rust-lang/rust/blob/17675855094906ba90aca2f119be8fb7afc4d456/src/bootstrap/check.rs#L507-L508

Stdout after removing `tool_check_step!(BootstrapShim...`:

```sh
  ~/devspace/.other/rust-dist/general  $ ~/devspace/ozkanonur/rust/x check
Building bootstrap
   Compiling bootstrap v0.0.0 (/home/nimda/devspace/ozkanonur/rust/src/bootstrap)
error[E0412]: cannot find type `BootstrapShim` in module `check`
   --> builder.rs:660:24
    |
660 |                 check::BootstrapShim,
    |                        ^^^^^^^^^^^^^
    |
   ::: check.rs:418:9
    |
418 |         pub struct $name {
    |         ---------------- similarly named struct `Bootstrap` defined here
    |
help: a struct with a similar name exists
    |
660 |                 check::Bootstrap,
    |                        ~~~~~~~~~
help: consider importing this struct
    |
1   | use crate::dist::BootstrapShim;
    |
help: if you import `BootstrapShim`, refer to it directly
    |
660 -                 check::BootstrapShim,
660 +                 BootstrapShim,
    |

For more information about this error, try `rustc --explain E0412`.
error: could not compile `bootstrap` due to previous error
failed to run: /home/nimda/devspace/.other/rust-dist/general/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --manifest-path /home/nimda/devspace/ozkanonur/rust/src/bootstrap/Cargo.toml
Build completed unsuccessfully in 0:00:01
```